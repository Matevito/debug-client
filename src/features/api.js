const axios = require('axios');

const api = axios.create({
    baseURL: "https://pure-falls-26749.herokuapp.com/apiv1/",
    timeout: 3000
});

export default api;