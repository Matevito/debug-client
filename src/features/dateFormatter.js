import { DateTime } from "luxon"

const simpleDate = (string) => {
    return DateTime.fromISO(string).toLocaleString(DateTime.DATE_MED)
};

const detailedDate = (string) => {
    return DateTime.fromISO(string).toLocaleString(DateTime.DATETIME_FULL_WITH_SECONDS)
}

export {
    simpleDate,
    detailedDate
}