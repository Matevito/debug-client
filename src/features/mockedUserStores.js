import configureStore from "redux-mock-store";

// todo: set project an issues values
const usersStoreValues = [
    {
        user: {} 
    },
    {
        user: {
            loggedIn: true,
            token : "a token?",
            user: {
                username: "devUser",
                email:"dev@email.com",
                role: "Developer",
                id: "1"
            },
            projects: {
                number: 0,
                list: []
            },
            issues: {
                number: 0,
                list: []
            }
        }
    },
    {

        user: {
            loggedIn: true,
            token : "a token?",
            user: {
                username: "TeamLeaderUser",
                email:"team@email.com",
                role: "Team leader",
                id: "2"
            },
            projects: {
                number: 0,
                list: []
            },
            issues: {
                number: 0,
                list: []
            }
        }
    },
    {

        user: {
            loggedIn: true,
            token : "admintoken?",
            user: {
                username: "adminUser",
                email:"admin@email.com",
                role: "Admin",
                id: "3"
            },
            projects: {
                number: 0,
                list: []
            },
            issues: {
                number: 0,
                list: []
            }
        }
    }
];

exports.getData = () => {
    const mockedUsers = usersStoreValues.map((user, i) => {
        const mockStore = configureStore([]);
        if (i === 0) {
            return mockStore(user)
        }
        return mockStore({user: {
            user: {
                user
            }
        }})
    });

    return mockedUsers;
}