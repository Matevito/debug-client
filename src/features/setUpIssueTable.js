const setUpIssueTable = (issuesList) => {
    let data = [
        {label: "open", x: 1,  y: 0},
        {label: "aditional info needed", x: 2, y: 0},
        {label: "in progress", x: 3, y: 0},
        {label: "under review", x: 4, y: 0},
        {label: "solved",x: 5, y: 0}
    ];

    issuesList.forEach((issue) => {
        data.forEach((obj, index) => {
            if (obj.label === issue.status){
                data[index].y += 1
            }
        })
    });
    
    return data.filter(obj => obj.y > 0);
};


export default setUpIssueTable;