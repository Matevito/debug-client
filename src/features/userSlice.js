import  { createSlice } from "@reduxjs/toolkit";

/* User object structure
user:{
    loggedIn: 'boolean',
    user: {
        username,
        email,
        role,
        id
    },
    projects: [{
        id,
        title
    }],
    issues: [{
        id,
        title
    }],
    todo on it's own slice:?
    notifications: [array of id's] // under development*
}
*/
export const userSlice = createSlice({
    name: 'user',
    initialState: {
        user: null
    },
    reducers: {
        login: (state, action) => {
            state.user = action.payload
        },
        logout: (state) => {
            state.user = null
        }
    }
});

export const { login, logout } = userSlice.actions;

export const selectUser = (state) => state.user.user;

export default userSlice.reducer