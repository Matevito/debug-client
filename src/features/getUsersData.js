import api from "./api";

const getUserData = async (token) => {
    const config = {
        headers: { "auth-token" : token}
    }; 
    try {
        const apiRes = await api.get("/user/list", config);
        return apiRes.data.data
    } catch(err) {
        return null
    }
}

export default getUserData