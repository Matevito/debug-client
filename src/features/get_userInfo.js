import api from "./api";

const get_userInfo = async (token) => {
    // setting up headers for calls
    const config = {
        headers: {
            "auth-token": token
        }
    };
    // fetch api data
    const userData = await api.get("/whoami", config);
    if (!userData) {
        return false;
    }
    const userId = userData.data.data.id;
    const userRelatedData = await api.get(`/user/${userId}`, config);
    if (!userRelatedData) {
        return false;
    };

    const userNotifications = [];

    // structure res and send it
    const dataRes = {
        loggedIn: true,
        token: token,
        user: {
            username: userData.data.data.username,
            email: userData.data.data.email,
            role: userData.data.data.role,
            id: userData.data.data.id
        },
        projects: userRelatedData.data.data.projects,
        issues: userRelatedData.data.data.issues,
        notifications: {
            list: userNotifications,
            number: userNotifications.length
        }
    };
    return dataRes;
}

export default get_userInfo;