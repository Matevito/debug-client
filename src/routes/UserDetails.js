import React, { useState, useEffect} from 'react'
import { useParams, useNavigate } from "react-router-dom";

// redux comp
import { useSelector } from "react-redux";
import { selectUser } from "../features/userSlice";

// mui comp
import {
    Box,
    Grid,
    Typography,
    Card,
    Button
} from "@mui/material";

// app comp
import { LoadingPage } from "../components/LoadingPage";

// api comp
import api from "../features/api";


export const UserDetails = () => {
    const user = useSelector(selectUser);
    const userId = useParams().id;
    let navigate = useNavigate();

    // state values;
    const [userInfo, setUserInfo] = useState(null);
    const [authorized, setAuthorized] = useState(null)

    // set-up user info
    useEffect(() => {
        async function getUserInfo () {
            const config = {
                headers: { "auth-token" : user.token }
            };
            try {
                const userRes = await api.get(`/user/${userId}`, config)
                setUserInfo(userRes.data.data);
                setAuthorized(true)
            } catch(err) {
                setAuthorized(false);
                navigate("/protected-route")
            }
        } 
        if (user) {
            if (user.user.id === userId) {
                navigate("/user/info");
            } else {
                getUserInfo()
            }
        }
    }, [user, userId, navigate]);

    // make user admin;

    if (!user) {
        navigate("/");
    } else if (authorized === null) {
        return(
            <LoadingPage />
        )
    } else if (authorized === true) {
        <>user displayed info</>
    }
}
