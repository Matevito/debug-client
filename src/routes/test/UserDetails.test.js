import { Provider } from "react-redux";
import { MemoryRouter, Route, Routes } from "react-router-dom";

// testing comps
import { render, screen, waitFor } from "@testing-library/react";
import '@testing-library/jest-dom'

// test comp
import { UserDetails } from "../UserDetails";

// mock functions;
import { getData } from "../../features/mockedUserStores";
import { rest } from "msw";
import { setupServer } from "msw/node";

const rootAPI = "https://pure-falls-26749.herokuapp.com/apiv1";
const UserResponse = {
    data: {
        user: {
            username: "adminUser",
            email: "admin@email.com",
            id: "adminID",
            role: "Admin"
        },
        projects: {list: [], number: 0},
        issues: {list: [{_id: "issueId", title: "testTicket"}], number: 1},
    }
}

const server = setupServer(
    rest.get(`${rootAPI}/user/adminID`, (req, res, ctx) => {
        return res(ctx.status(200), ctx.json(UserResponse));
    }),
    rest.get(`${rootAPI}/user/devID`, (req, res, ctx) => {
        return res(ctx.status(200), ctx.json({
            data: {
                user: {
                    username: "developer",
                    email: "developer@email.arg",
                    id: "devID",
                    role:"Developer"
                },
                projects: {list: [], number: 0},
                issues: {list: [{_id: "issueId", title: "testTicket"}], number: 1},
            }
        }))
    })
);
const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
}));

const renderComponent = (url, store) => {
    return render(
        <Provider store={store}>
            <MemoryRouter initialEntries={[url]}>
                <Routes>
                    <Route 
                        path="/user/:id"
                        element={<UserDetails />}
                    />
                </Routes>
            </MemoryRouter>
        </Provider>
    )
}
describe("UserDetails component", () => {
    // test requirements
    let noLoggedUser;
    let devStore;
    let adminStore;
    beforeEach(() => {
        const usersData = getData();
        noLoggedUser = usersData[0];
        devStore = usersData[1];
        adminStore = usersData[3];
    });
    beforeAll(() => server.listen())
    afterEach(() => server.resetHandlers());
    afterAll(() => {
        server.close()
        jest.clearAllMocks();
    });

    // authentication tests
    test("handles no user set-up", () => {
        const testURL = "/user/testing"
        renderComponent(testURL, noLoggedUser);
        expect(mockedUsedNavigate).toHaveBeenCalledWith("/")
    });
    test("handles if the api response is negative", async() => {
        const testURL = "/user/rejectThis";
        server.use(
            rest.get(`${rootAPI}/user/rejectThis`, (req, res, ctx) => {
                return res(ctx.status(400))
            })
        );
        renderComponent(testURL, devStore);
        await waitFor(() => mockedUsedNavigate.mock.lastCall[0]);
        expect(mockedUsedNavigate).toHaveBeenCalledWith("/protected-route")
    });

    test("handles if the requested user is the one making the call", async() => {
        const testURL = "/user/1";
        renderComponent(testURL, devStore)
        await waitFor(() => mockedUsedNavigate.mock.lastCall[0]);
        expect(mockedUsedNavigate).toHaveBeenCalledWith("/user/info")
    })
    // functional tests;
    test("renders admin page", async() => {
        const testURL = "/user/adminID";
        const { container } = renderComponent(testURL, devStore)
        await screen.findByTestId("GroupsIcon");
        expect(container).toMatchSnapshot();
    });
    test("renders no admin page", async() => {
        const testURL = "/user/devID";
        const { container } = renderComponent(testURL, adminStore)
        await screen.findByTestId("GroupsIcon");
        expect(container).toMatchSnapshot();
    });
    test("handles makeAdmin btn", async() => {
        const testURL = "/user/devID";
        renderComponent(testURL, adminStore)
        await screen.findByTestId("GroupsIcon");
        const adminBtn = screen.getByText("make admin")
        expect(adminBtn).toBeInTheDocument()
    })
})