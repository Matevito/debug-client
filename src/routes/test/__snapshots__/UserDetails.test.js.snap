// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails component renders admin page 1`] = `
<div>
  <div
    class="MuiBox-root css-af1f4c"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-1r7yq5s-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-1vjpr1i-MuiCardContent-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AccountBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
          />
        </svg>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          ADMINUSER
        </h3>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-2bmocg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hn59hz-MuiGrid-root"
          >
            <div
              class="css-1kq5d21-MuiStack-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BadgeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 7h-5V4c0-1.1-.9-2-2-2h-2c-1.1 0-2 .9-2 2v3H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zM9 12c.83 0 1.5.67 1.5 1.5S9.83 15 9 15s-1.5-.67-1.5-1.5S8.17 12 9 12zm3 6H6v-.75c0-1 2-1.5 3-1.5s3 .5 3 1.5V18zm1-9h-2V4h2v5zm5 7.5h-4V15h4v1.5zm0-3h-4V12h4v1.5z"
                />
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                <b>
                  User role
                </b>
              </h6>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Admin
            </p>
            <div
              class="css-1kq5d21-MuiStack-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EmailIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                <b>
                  Contact email
                </b>
              </h6>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              admin@email.com
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hn59hz-MuiGrid-root"
          >
            <div
              class="css-147rfyl-MuiStack-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="GroupsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12.75c1.63 0 3.07.39 4.24.9 1.08.48 1.76 1.56 1.76 2.73V18H6v-1.61c0-1.18.68-2.26 1.76-2.73 1.17-.52 2.61-.91 4.24-.91zM4 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm1.13 1.1c-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58C.48 14.9 0 15.62 0 16.43V18h4.5v-1.61c0-.83.23-1.61.63-2.29zM20 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85-.85-.37-1.79-.58-2.78-.58-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V18H24v-1.57zM12 6c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3z"
                />
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                <b>
                  Projects user is part 
                </b>
              </h6>
              <h5
                class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              >
                0
              </h5>
            </div>
            <div
              class="css-147rfyl-MuiStack-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BugReportIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20V8zm-6 8h-4v-2h4v2zm0-4h-4v-2h4v2z"
                />
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                <b>
                  Tickets taken
                </b>
              </h6>
              <h5
                class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              >
                1
              </h5>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserDetails component renders no admin page 1`] = `
<div>
  <div
    class="MuiBox-root css-af1f4c"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-1r7yq5s-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-1vjpr1i-MuiCardContent-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AccountBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
          />
        </svg>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          DEVELOPER
        </h3>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-2bmocg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hn59hz-MuiGrid-root"
          >
            <div
              class="css-1kq5d21-MuiStack-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BadgeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 7h-5V4c0-1.1-.9-2-2-2h-2c-1.1 0-2 .9-2 2v3H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zM9 12c.83 0 1.5.67 1.5 1.5S9.83 15 9 15s-1.5-.67-1.5-1.5S8.17 12 9 12zm3 6H6v-.75c0-1 2-1.5 3-1.5s3 .5 3 1.5V18zm1-9h-2V4h2v5zm5 7.5h-4V15h4v1.5zm0-3h-4V12h4v1.5z"
                />
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                <b>
                  User role
                </b>
              </h6>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Developer
            </p>
            <div
              class="css-1kq5d21-MuiStack-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EmailIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                <b>
                  Contact email
                </b>
              </h6>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              developer@email.arg
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hn59hz-MuiGrid-root"
          >
            <div
              class="css-147rfyl-MuiStack-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="GroupsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12.75c1.63 0 3.07.39 4.24.9 1.08.48 1.76 1.56 1.76 2.73V18H6v-1.61c0-1.18.68-2.26 1.76-2.73 1.17-.52 2.61-.91 4.24-.91zM4 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm1.13 1.1c-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58C.48 14.9 0 15.62 0 16.43V18h4.5v-1.61c0-.83.23-1.61.63-2.29zM20 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85-.85-.37-1.79-.58-2.78-.58-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V18H24v-1.57zM12 6c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3z"
                />
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                <b>
                  Projects user is part 
                </b>
              </h6>
              <h5
                class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              >
                0
              </h5>
            </div>
            <div
              class="css-147rfyl-MuiStack-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BugReportIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20V8zm-6 8h-4v-2h4v2zm0-4h-4v-2h4v2z"
                />
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                <b>
                  Tickets taken
                </b>
              </h6>
              <h5
                class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              >
                1
              </h5>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-4hrfbl-MuiCardActions-root"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-1osjee8-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EngineeringIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4zm13.1-8.16c.01-.11.02-.22.02-.34 0-.12-.01-.23-.03-.34l.74-.58c.07-.05.08-.15.04-.22l-.7-1.21c-.04-.08-.14-.1-.21-.08l-.86.35c-.18-.14-.38-.25-.59-.34l-.13-.93c-.02-.09-.09-.15-.18-.15h-1.4c-.09 0-.16.06-.17.15l-.13.93c-.21.09-.41.21-.59.34l-.87-.35c-.08-.03-.17 0-.21.08l-.7 1.21c-.04.08-.03.17.04.22l.74.58c-.02.11-.03.23-.03.34 0 .11.01.23.03.34l-.74.58c-.07.05-.08.15-.04.22l.7 1.21c.04.08.14.1.21.08l.87-.35c.18.14.38.25.59.34l.13.93c.01.09.08.15.17.15h1.4c.09 0 .16-.06.17-.15l.13-.93c.21-.09.41-.21.59-.34l.87.35c.08.03.17 0 .21-.08l.7-1.21c.04-.08.03-.17-.04-.22l-.73-.58zm-2.6.91c-.69 0-1.25-.56-1.25-1.25s.56-1.25 1.25-1.25 1.25.56 1.25 1.25-.56 1.25-1.25 1.25zm.42 3.93-.5-.87c-.03-.06-.1-.08-.15-.06l-.62.25c-.13-.1-.27-.18-.42-.24l-.09-.66c-.02-.06-.08-.1-.14-.1h-1c-.06 0-.11.04-.12.11l-.09.66c-.15.06-.29.15-.42.24l-.62-.25c-.06-.02-.12 0-.15.06l-.5.87c-.03.06-.02.12.03.16l.53.41c-.01.08-.02.16-.02.24 0 .08.01.17.02.24l-.53.41c-.05.04-.06.11-.03.16l.5.87c.03.06.1.08.15.06l.62-.25c.13.1.27.18.42.24l.09.66c.01.07.06.11.12.11h1c.06 0 .12-.04.12-.11l.09-.66c.15-.06.29-.15.42-.24l.62.25c.06.02.12 0 .15-.06l.5-.87c.03-.06.02-.12-.03-.16l-.52-.41c.01-.08.02-.16.02-.24 0-.08-.01-.17-.02-.24l.53-.41c.05-.04.06-.11.04-.17zm-2.42 1.65c-.46 0-.83-.38-.83-.83 0-.46.38-.83.83-.83s.83.38.83.83c0 .46-.37.83-.83.83zM4.74 9h8.53c.27 0 .49-.22.49-.49v-.02c0-.27-.22-.49-.49-.49H13c0-1.48-.81-2.75-2-3.45v.95c0 .28-.22.5-.5.5s-.5-.22-.5-.5V4.14C9.68 4.06 9.35 4 9 4s-.68.06-1 .14V5.5c0 .28-.22.5-.5.5S7 5.78 7 5.5v-.95C5.81 5.25 5 6.52 5 8h-.26c-.27 0-.49.22-.49.49v.03c0 .26.22.48.49.48zM9 13c1.86 0 3.41-1.28 3.86-3H5.14c.45 1.72 2 3 3.86 3z"
              />
            </svg>
          </span>
          make admin
        </button>
      </div>
    </div>
  </div>
</div>
`;
